### ISSUE TITLE
How to enable progress bar in code.

### ISSUE THREAD

**Issue by zh-plus** [m-bain/whisperX#327]
We can output the progress bar using `--verbose` in cli.

However, can I enable the progress bar when invoking function?

For examle:

```python
whisper_model = whisperx.load_model('large-v2', compute_type='float16', device=self.device)
audio = whisperx.load_audio(audio_path)

result = whisper_model.transcribe(audio, batch_size=batch_size)

align_model, metadata = whisperx.load_align_model(language_code=result["language"], device=self.device)aligned_result = whisperx.align(result["segments"], align_model, metadata, audio, self.device, return_char_alignments=False)
```

It would be beneficial to include an additional parameter `progress=True` for both the functions `whisper_model.transcribe` and `whisperx.align`. This will prove useful when dealing with lengthy audio files, as it can be frustrating to simply wait without any indication of progress.

**Comment by kaanchan** [comment]
> We can output the progress bar using `--verbose` in cli.

@zh-plus Can you provide a working example of this? I am unable to reproduce in my installation. Using "--verbose True" did not show a progress bar. 

The full command I used was `whisperx '.\Yoga Nidra in English.mp3' --model large-v2 --align_model WAV2VEC2_ASR_LARGE_LV60K_960H --batch_size 4 --compute_type float32 --highlight_words True --verbose True`

**Comment by anilbey** [comment]
Setting `--verbose True` also does not show a progress bar. In fact `--verbose` is set `True` by default but there is no progress bar. 

The only verbosity it provides during the transcription is the following:

> `>>Performing transcription...`

**Comment by Sogl** [comment]
Need progress and phrase output as in whisper.cpp

**Comment by kirankunigiri** [comment]
Has anyone found a solution for this? The original faster whisper library shows a progress bar. Also, inside the transcribe.py file of this repo, I see code for showing a progress bar. This comment exists in that file: 
`# show the progress bar when verbose is False (if True, transcribed text will be printed)`
And there is code from the progress bar library tqdm to show it. However, regardless of if `verbose` is true or false, nothing is ever printed. Is there some other setting that needs to be changed?

**Comment by Infinitay** [comment]
I agree that a way to tell the progress of all the various functions would be nice. Personally, I modified whisperx locally and built it so that I have more knowledge on how the progress is going in the form of basic print statements for when transcription/alignment/diarization starts and finishes. Although it's primitive in that it doesn't indicate true progress of the inferencing.

**Comment by Barabazs** [comment]
Closing as duplicate of #220. print_progress is now available as a parameter for both transcribe() and align() in the Python API.
